### PR TITLE
I2C Sem4

### PR DESCRIPTION
**Pull request by manthanand**
This branch is now done

### CODE REVIEW

**Comment by manthanand** [comment]
Since we have only the EEPROM in our I2C, I removed the parameter in the functions that pass in the address of the EEPROM since it will always be the same. The reason it can't build is because I changed the header file.

**Comment by ClarkPoon** [comment]
@manthanand can you fix the build errors?

**Comment by manthanand** [comment]
The reason for the build errors is because the ifdef's in the file are not defined, so all these static functions are defined at the top of the file but there are no functions underneath. This could be fixed by adding a #define at the top of the file or compiling with the command you told me about. I thought we didn't want #defines at the top of the file so I removed them.

**Comment by ClarkPoon** [comment]
> The reason for the build errors is because the ifdef's in the file are not defined, so all these static functions are defined at the top of the file but there are no functions underneath. This could be fixed by adding a #define at the top of the file or compiling with the command you told me about. I thought we didn't want #defines at the top of the file so I removed them.

Ok. I think we should either add in a `#define RTOS` or add the compiler option to define it to the makefile. What do you think @manthanand @chaseblock @SijWoo @sugamarora99

**Comment by manthanand** [comment]
I think making it a compiler option would be better. Otherwise we would have to change it for every file that uses these #defines (UART, SPI so far).

**Comment by ClarkPoon** [comment]
> I think making it a compiler option would be better. Otherwise we would have to change it for every file that uses these #defines (UART, SPI so far).

If we added it as a `#define` we could just have one in config.h and `#include "config.h"`  in all of the files that need it

**Comment by manthanand** [comment]
Oh, but then how would we build both versions in GitHub (if you want to make sure both versions work like we do for simulator and STM)?

**Comment by ClarkPoon** [comment]
> Oh, but then how would we build both versions in GitHub (if you want to make sure both versions work like we do for simulator and STM)?

To build both versions in GitHub we will probably have to edit our testing script and pass in RTOS and BAREMETAL as compiler options

File: BSP/Inc/BSP_I2C.h [hunk 1]
```diff
@@ -14,22 +15,22 @@ void BSP_I2C_Init(void);
 
 /**
  * @brief   Transmits data onto the I2C bus.
- * @param   deviceAddr : the device/IC that the data is intended for.
+ * @param   devAddr : the device address to write to
```
Comment (ClarkPoon) [inline BSP/Inc/BSP_I2C.h:18]: Why did we change the parameter name from `deviceAddr` to `devAddr` the first one seems better to me.

**Review by sugamarora99 — DISMISSED** [review]

**Review by ClarkPoon — APPROVED** [review]

**Review by ClarkPoon — APPROVED** [review]

**Pull request merged**